File: sources/net.sf.j2s.core/test/dev/j2s/org/sun/font/EAttribute.js
Clazz.declarePackage ("sun.font");
Clazz.load (["java.lang.Enum"], "sun.font.EAttribute", ["java.awt.font.TextAttribute"], function () {
c$ = Clazz.decorateAsClass (function () {
this.mask = 0;
this.att = null;
Clazz.instantialize (this, arguments);
}, sun.font, "EAttribute", Enum);
Clazz.makeConstructor (c$, 
function (ta) {
this.mask = 1 << this.ordinal ();
this.att = ta;
}, "java.awt.font.TextAttribute");
c$.forAttribute = Clazz.defineMethod (c$, "forAttribute", 
function (ta) {
for (var ea, $ea = 0, $$ea = sun.font.EAttribute.atts; $ea < $$ea.length && ((ea = $$ea[$ea]) || true); $ea++) {
if (ea.att === ta) {
return ea;
}}
return null;
}, "java.text.AttributedCharacterIterator.Attribute");
Clazz.overrideMethod (c$, "toString", 
function () {
return this.name ().substring (1).toLowerCase ();
});
c$.atts = sun.font.EAttribute.getEnumConstants ();
Clazz.defineEnumConstant (c$, "EFAMILY", 0, [java.awt.font.TextAttribute.FAMILY]);
Clazz.defineEnumConstant (c$, "EWEIGHT", 1, [java.awt.font.TextAttribute.WEIGHT]);
Clazz.defineEnumConstant (c$, "EWIDTH", 2, [java.awt.font.TextAttribute.WIDTH]);
Clazz.defineEnumConstant (c$, "EPOSTURE", 3, [java.awt.font.TextAttribute.POSTURE]);
Clazz.defineEnumConstant (c$, "ESIZE", 4, [java.awt.font.TextAttribute.SIZE]);
Clazz.defineEnumConstant (c$, "ETRANSFORM", 5, [java.awt.font.TextAttribute.TRANSFORM]);
Clazz.defineEnumConstant (c$, "ESUPERSCRIPT", 6, [java.awt.font.TextAttribute.SUPERSCRIPT]);
Clazz.defineEnumConstant (c$, "EFONT", 7, [java.awt.font.TextAttribute.FONT]);
Clazz.defineEnumConstant (c$, "ECHAR_REPLACEMENT", 8, [java.awt.font.TextAttribute.CHAR_REPLACEMENT]);
Clazz.defineEnumConstant (c$, "EFOREGROUND", 9, [java.awt.font.TextAttribute.FOREGROUND]);
Clazz.defineEnumConstant (c$, "EBACKGROUND", 10, [java.awt.font.TextAttribute.BACKGROUND]);
Clazz.defineEnumConstant (c$, "EUNDERLINE", 11, [java.awt.font.TextAttribute.UNDERLINE]);
Clazz.defineEnumConstant (c$, "ESTRIKETHROUGH", 12, [java.awt.font.TextAttribute.STRIKETHROUGH]);
Clazz.defineEnumConstant (c$, "ERUN_DIRECTION", 13, [java.awt.font.TextAttribute.RUN_DIRECTION]);
Clazz.defineEnumConstant (c$, "EBIDI_EMBEDDING", 14, [java.awt.font.TextAttribute.BIDI_EMBEDDING]);
Clazz.defineEnumConstant (c$, "EJUSTIFICATION", 15, [java.awt.font.TextAttribute.JUSTIFICATION]);
Clazz.defineEnumConstant (c$, "EINPUT_METHOD_HIGHLIGHT", 16, [java.awt.font.TextAttribute.INPUT_METHOD_HIGHLIGHT]);
Clazz.defineEnumConstant (c$, "EINPUT_METHOD_UNDERLINE", 17, [java.awt.font.TextAttribute.INPUT_METHOD_UNDERLINE]);
Clazz.defineEnumConstant (c$, "ESWAP_COLORS", 18, [java.awt.font.TextAttribute.SWAP_COLORS]);
Clazz.defineEnumConstant (c$, "ENUMERIC_SHAPING", 19, [java.awt.font.TextAttribute.NUMERIC_SHAPING]);
Clazz.defineEnumConstant (c$, "EKERNING", 20, [java.awt.font.TextAttribute.KERNING]);
Clazz.defineEnumConstant (c$, "ELIGATURES", 21, [java.awt.font.TextAttribute.LIGATURES]);
Clazz.defineEnumConstant (c$, "ETRACKING", 22, [java.awt.font.TextAttribute.TRACKING]);
Clazz.defineEnumConstant (c$, "EBASELINE_TRANSFORM", 23, [null]);
});

File: sources/net.sf.j2s.java.core/src/sun/font/EAttribute.java
/*
 * Some portions of this file have been modified by Robert Hanson hansonr.at.stolaf.edu 2012-2017
 * for use in SwingJS via transpilation into JavaScript using Java2Script.
 *
 * Copyright (c) 2005, Oracle and/or its affiliates. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.  Oracle designates this
 * particular file as subject to the "Classpath" exception as provided
 * by Oracle in the LICENSE file that accompanied this code.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
 * or visit www.oracle.com if you need additional information or have any
 * questions.
 */

/*
 * (C) Copyright IBM Corp. 2005 - All Rights Reserved
 *
 * The original version of this source code and documentation is
 * copyrighted and owned by IBM. These materials are provided
 * under terms of a License Agreement between IBM and Sun.
 * This technology is protected by multiple US and International
 * patents. This notice and attribution to IBM may not be removed.
 */

package sun.font;

import static java.awt.font.TextAttribute.BACKGROUND;
import static java.awt.font.TextAttribute.BIDI_EMBEDDING;
import static java.awt.font.TextAttribute.CHAR_REPLACEMENT;
import static java.awt.font.TextAttribute.FAMILY;
import static java.awt.font.TextAttribute.FONT;
import static java.awt.font.TextAttribute.FOREGROUND;
import static java.awt.font.TextAttribute.INPUT_METHOD_HIGHLIGHT;
import static java.awt.font.TextAttribute.INPUT_METHOD_UNDERLINE;
import static java.awt.font.TextAttribute.JUSTIFICATION;
import static java.awt.font.TextAttribute.KERNING;
import static java.awt.font.TextAttribute.LIGATURES;
import static java.awt.font.TextAttribute.NUMERIC_SHAPING;
import static java.awt.font.TextAttribute.POSTURE;
import static java.awt.font.TextAttribute.RUN_DIRECTION;
import static java.awt.font.TextAttribute.SIZE;
import static java.awt.font.TextAttribute.STRIKETHROUGH;
import static java.awt.font.TextAttribute.SUPERSCRIPT;
import static java.awt.font.TextAttribute.SWAP_COLORS;
import static java.awt.font.TextAttribute.TRACKING;
import static java.awt.font.TextAttribute.TRANSFORM;
import static java.awt.font.TextAttribute.UNDERLINE;
import static java.awt.font.TextAttribute.WEIGHT;
import static java.awt.font.TextAttribute.WIDTH;

import java.text.AttributedCharacterIterator.Attribute;

import java.awt.font.TextAttribute;

public enum EAttribute {
    EFAMILY(FAMILY),
    EWEIGHT(WEIGHT),
    EWIDTH(WIDTH),
    EPOSTURE(POSTURE),
    ESIZE(SIZE),
    ETRANSFORM(TRANSFORM),
    ESUPERSCRIPT(SUPERSCRIPT),
    EFONT(FONT),
    ECHAR_REPLACEMENT(CHAR_REPLACEMENT),
    EFOREGROUND(FOREGROUND),
    EBACKGROUND(BACKGROUND),
    EUNDERLINE(UNDERLINE),
    ESTRIKETHROUGH(STRIKETHROUGH),
    ERUN_DIRECTION(RUN_DIRECTION),
    EBIDI_EMBEDDING(BIDI_EMBEDDING),
    EJUSTIFICATION(JUSTIFICATION),
    EINPUT_METHOD_HIGHLIGHT(INPUT_METHOD_HIGHLIGHT),
    EINPUT_METHOD_UNDERLINE(INPUT_METHOD_UNDERLINE),
    ESWAP_COLORS(SWAP_COLORS),
    ENUMERIC_SHAPING(NUMERIC_SHAPING),
    EKERNING(KERNING),
    ELIGATURES(LIGATURES),
    ETRACKING(TRACKING),
    EBASELINE_TRANSFORM(null);

    /* package */ final int mask;
    /* package */ final TextAttribute att;

    EAttribute(TextAttribute ta) {
        mask = 1 << ordinal();
        att = ta;
    }

    /* package */ static final EAttribute[] atts = EAttribute.class.getEnumConstants();

    public static EAttribute forAttribute(Attribute ta) {
        for (EAttribute ea: atts) {
            if (ea.att == ta) {
                return ea;
            }
        }
        return null;
    }

    @Override
		public String toString() {
        return name().substring(1).toLowerCase();
    }
}
